### DISCUSSION TITLE
[Feature]: Autosave only when the session file of current working directory exists

### DISCUSSION THREAD

**Discussion by MasouShizuka** [olimorris/persisted.nvim#95] [Ideas]
This feature is similar to `autosave_only_in_session` of the plugin `neovim-session-manager`.
With this feature, I can autoread the session of only my manually saved working directory.

Although there is currently a setting of `should_autosave`, I cannot determine through the API whether there is a session of the current working directory in the session directory.
At present, it seems that can only compare each `file_path` with the current working path through `list()`
However, the naming of the session file may change with plug-in updates.

Therefore, perhaps an API should be provided to determine whether the session file of the current directory exists?

**Comment by MasouShizuka** [comment]
Currently, if I want to implement this feature, I must write in config:
```lua
local persisted = require("persisted")

local save_dir = vim.fn.expand(variables.data_path .. "/lazy/persisted.nvim/sessions/")
opts["save_dir"] = save_dir
opts["should_autosave"] = function()
    local name = vim.fn.getcwd():gsub(require("persisted.utils").get_dir_pattern(), "%%")
    local session = save_dir .. name .. persisted.get_branch() .. ".vim"
    return vim.fn.filereadable(session) ~= 0
end

persisted.setup(opts)
```

In order for this to work, it must be consistent with the repository's code.
So I still hope to be able to provide this feature.

EDIT: Oh, I found that `SessionSave` must meet the conditions of `should_autosave`, so it cannot be saved in the new working directory. It can only update existing session files.

**Reply by olimorris** [reply]
@MasouShizuka sorry my time is quite constrained these days. I'd be willing to accept a PR if you wish to add any functionality that solves this.
